sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1708,6 +2047,275 @@ a normal SYCL command-group submission.
+----
+
+|Overload of `depends_on()` which takes a list of dynamic event dependencies.
+
+This function has the same semantics as `depends_on(const std::vector<event>&)`.
+
+Parameters:
+
+* `depEvents` - List of dynamic event dependencies.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if this function is called from
+a normal SYCL command-group submission.
+
+* Throws synchronously with error code `invalid` if any of the `sycl::event`
+objects associated with `depEvents` came from the same graph that the graph node
+resulting from this command-group submission is associated with.
+
+* Throws synchronously with error code `invalid` if any of `depEvents` is a
+default constructed `dynamic_event` with no underlying SYCL event.
+|===
+
+=== Events
+
+Events can be used with graphs in the following ways:
+
+- Defining dependencies between nodes in the same graph.
+- Defining dependencies between <<inter-graph-dependencies, different graphs>>.
+- Obtaining <<node-execution-events, execution events for invidual nodes>>
+within an executable graph, which can be waited on or used as dependencies for
+eager SYCL operations.
+- Creating external event dependencies for a graph, either
+<<external-event-dependencies, one-time>> or <<dynamic-events, updatable between
+executions>>.
+
+==== Node-Level Execution Events [[node-execution-events]]
+
+Events representing the completion of an individual node inside a given
+executable graph can be obtained using
+`command_graph<graph_state::executable>::get_event(node)`. These events can then
+be waited on or used as dependencies for eager SYCL operations outside of
+graphs. These events may be useful for operations which may be infrequent and
+depend only on some intermediate results of work being done in the graph.
+
+[source, c++]
+....
+sycl::event ExecutionEvent = ExecGraph.get_event(SomeNode);
+
+Queue.submit((sycl::handler& CGH)
+  {
+    CGH.depends_on(ExecutionEvent);
+    CGH.parallel_for(...);
+  });
+
+// The above operation will only execute once SomeNode has finished executing
+// inside ExecGraph
+Queue.ext_oneapi_graph(ExecGraph);
+....
+
+These events represent only the most recent execution of a given executable
+graph. If an application executes the same graph multiple times before
+scheduling work or performing a host-side wait on the event then executions of
+the node in a previous execution other than the most recent one may be missed.
+Applications requiring this should take care to schedule eager operations/waits
+between each graph execution, or include these operations as nodes in the graph
+if they are to be performed for every graph execution.
+
+These events cannot be used to define dependencies between graphs. These should
+instead be defined as described in <<inter-graph-dependencies, this section>>.
+
+==== Adding External Event Dependencies To Graphs [[external-event-dependencies]]
+
+<<event-terminology, Regular SYCL events>> can be passed as dependencies to
```

Comment:
Regular SYCL events include:
1) normal submissions to a SYCL queue
2) events returned from submitting an executable command_graph for execution
3) events obtained via command_graph<graph_state::executable>::get_event()

as for item 3, `These events can then be waited on or used as dependencies for eager SYCL operations **outside** of graphs`, so can item 3 be used as `dependencies to graph nodes`?

Reply:
No this is explicitly disallowed in the spec, dependencies between graph nodes should be established using graph-limited events.

Reply:
If the answer is no, I think we can not just use "Regular SYCL events" here since not all the three parts are all allowed.